wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,344 @@
+            except (discord.HTTPException, discord.Forbidden, discord.InvalidArgument):
+                dm = False
+                logger.info("[Ban ban()] Notification dm to user failed due to user preferences")
+
+            # kick
+            await user.kick(reason=reason)
+            dt = datetime.datetime.now(pytz.utc)
+            logger.info(f"[Ban ban()] User kicked from guiled at {dt}.")
+
+            # report to council
+            e_obj = discord.Embed(title="Ban Hammer Deployed",
+                                  colour=discord.Color.red())
+
+            e_obj.add_field(name="Banned User", value=f"**{username}**", inline=True)
+            e_obj.add_field(name="Moderator", value=f"**{mod_info[0]}**", inline=True)
+            e_obj.add_field(name="Reason", value=f"```{reason}```", inline=False)
+            e_obj.add_field(name="Notification DM", value="SENT\n" if dm else "NOT SENT, DUE TO USER DM PREF's\n",
+                            inline=False)
+            e_obj.set_footer(text="Moderator Action")
+            if e_obj:
+                e_obj.timestamp = dt.astimezone(pytz.timezone('Canada/Pacific'))
+                await self.mod_channel.send(embed=e_obj)
+            logger.info(f"[Ban ban()] Message sent to mod channel,{self.mod_channel}, of the ban for {username}.")
+
+            # update db
+            await BannedUsers.insert_ban(username, user.id)
+            await BanRecords.insert_record(username, user.id, mod_info[0], mod_info[1], dt, reason)
+
+    @commands.command()
+    async def unban(self, ctx, _id: int):
+        """Unbans a user"""
+
+        logger.info(f"[Ban unban()] unban command detected from {ctx.author} with args=[ {_id} ]")
+        if _id not in self.ban_list:
+            logger.info(f"Provided id: {_id}, does not belong to a banned member.")
+            e_obj = await em(ctx, title="Error",
+                             content=[("Problem",
+                                       f"`{_id}` is either not a valid Discord ID **OR** is not a banned user.")],
+                             colour=self.errorColour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+            return
+
+        # "unban"
+        self.ban_list.remove(_id)
+
+        name = await BannedUsers.del_banned_user_by_id(_id)
+
+        logger.info(f"[Ban unban()] User: {name} with id: {_id} was unbanned.")
+        e_obj = await em(ctx, title="Unban", description=f"**`{name}`** was unbanned.", colour=discord.Color.red())
+        if e_obj:
+            await self.mod_channel.send(embed=e_obj)
+
+    @unban.error
+    async def unban_error(self, ctx, error):
+        """Catches an error in unban when a non integer is passed in as an argument"""
+
+        logger.info("[Ban Unban_error] caught non integer ID passed into unban parameter. Handled accordingly")
+        if isinstance(error, commands.BadArgument):
```

Comment:
why only print out an error message if the error is of `BadArgument` type ?

Reply:
only error i cared about. if the user passes in a non it its a badargument. anything else i doubt will happen and if it does something else will catch it like for any other command.